src/Tools/dotnet-trace/DiagnosticProfiles.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.Diagnostics.Tools.Trace
+{
+    internal class DiagnosticProfileProvider
```

Comment:
In the same vein as the user-experience comment that we should have one notion of 'Profile', hopefully we only need one set of implementation types for these concepts as well. Could we use EventPipeProvider in place of DiagnosticProfileProvider and [Profile](https://github.com/dotnet/diagnostics/blob/master/src/Tools/dotnet-trace/Profile.cs#L11) in place of DiagnosticProfile?